### PR TITLE
Fix zero rigid body mass

### LINKED ISSUE #526: Attempting to set Rigid Body Mass to 0 causes crash.
How to recreate:
1. Add a Rigid Body component.
2. Manually type in 0 or a negative number in 'Mass'.

### PR DESCRIPTION
**Pull request by andens**
Fixes #526 

The problem was not only with 0 or negative mass, but any mass when no shape component was present. Setting mass also alters inertia, which requires the shape. Since no shape was ever set, a dereference of a null pointer followed. We assert for now that a rigid body without shape makes no sense, so attempting to set mass is silently ignored if no shape is present. To make things intuitive for level designers, you will be presented a helpful message in the editor and you won't even be able to set stuff on the rigid body until you have also added a shape component.

Still, setting negative mass makes no sense (or does it? :dancing_women:), so I'll open a bug to fix this too.

### CODE REVIEW

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![INFO][INFO] [EntityEditor.cpp#L450](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/b26b2d93e802c685eff9b8b0ce00a83b980e570e/src/Editor/GUI/Editors/EntityEditor.cpp#L450): keyword 'if' not followed by a single space [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT008)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Editor/GUI/Editors/EntityEditor.cpp [hunk 1]
```diff
@@ -349,11 +349,18 @@ void EntityEditor::ListenerEditor(Component::Listener* listener) {
 }
 
 void EntityEditor::RigidBodyEditor(Component::RigidBody* rigidBody) {
-    ImGui::Indent();
-    if (ImGui::InputFloat("Mass", &rigidBodyMass)) {
-        Managers().physicsManager->SetMass(rigidBody, rigidBodyMass);
+    auto shapeComp = rigidBody->entity->GetComponent<Component::Shape>();
+    if (shapeComp) {
+        ImGui::Indent();
+        if (ImGui::InputFloat("Mass", &rigidBodyMass)) {
```
Comment (Chainsawkitten) [inline src/Editor/GUI/Editors/EntityEditor.cpp]: Single line ifs without brackets.

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![INFO][INFO] [EntityEditor.cpp#L450](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/061990491d10421f9c2ee53121e6ed6008755034/src/Editor/GUI/Editors/EntityEditor.cpp#L450): keyword 'if' not followed by a single space [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT008)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![INFO][INFO] [EntityEditor.cpp#L450](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/19d76515f333c10f20786832f1068639fc77295c/src/Editor/GUI/Editors/EntityEditor.cpp#L450): keyword 'if' not followed by a single space [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT008)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**